sentry-android-core/src/main/java/io/sentry/android/core/internal/util/SentryFrameMetricsCollector.java

```diff
@@ -83,18 +89,83 @@ public SentryFrameMetricsCollector(
     // start a profile, we wouldn't have the current activity and couldn't get the frameMetrics.
     ((Application) context).registerActivityLifecycleCallbacks(this);
 
+    // Most considerations regarding timestamps of frames are inspired from JankStats library:
+    // https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:metrics/metrics-performance/src/main/java/androidx/metrics/performance/JankStatsApi24Impl.kt
+
+    // The Choreographer instance must be accessed on the main thread
+    new Handler(Looper.getMainLooper()).post(() -> choreographer = Choreographer.getInstance());
+    // Let's get the last frame timestamp from the choreographer private field
+    try {
+      choreographerLastFrameTimeField = Choreographer.class.getDeclaredField("mLastFrameTimeNanos");
+      choreographerLastFrameTimeField.setAccessible(true);
+    } catch (NoSuchFieldException e) {
+      options
+          .getLogger()
+          .log(SentryLevel.ERROR, "Unable to get the frame timestamp from the choreographer: ", e);
+    }
     frameMetricsAvailableListener =
         (window, frameMetrics, dropCountSinceLastInvocation) -> {
-          float refreshRate =
+          final long now = System.nanoTime();
+          final float refreshRate =
               buildInfoProvider.getSdkInfoVersion() >= Build.VERSION_CODES.R
                   ? window.getContext().getDisplay().getRefreshRate()
                   : window.getWindowManager().getDefaultDisplay().getRefreshRate();
+
+          final long cpuDuration = getFrameCpuDuration(frameMetrics);
+          long startTime = getFrameStartTimestamp();
+          // If we couldn't get the timestamp through reflection, we use current time
+          if (startTime == -1) {
```

Comment:
`m`: would it be possible to test this new logic (and everything below that)? I see we test that we access Choreographer, but this would also worth to test I think.

Reply:
added the new tests